src/common/transformations/src/transformations/symbolic_transformations/symbol_optimization.cpp

```diff
@@ -218,7 +219,101 @@ void optimize_value_usage(ov::Output<ov::Node>& output, STS_map& symbol_shape_so
     }
 }
 
-void save_shape_sources(const ov::Output<ov::Node>& output, STS_map& symbol_shape_source) {
+std::vector<std::shared_ptr<ov::Node>> topological_order(const std::shared_ptr<ov::Model>& m) {
+    auto order = m->get_ordered_ops();
+
+    // step 1: split model into parameter related and parameter non-related ops
+    const std::string op_depends_on_parameter = "topological_sort_op_depends_on";
+    // values: true - parameter dependent; false otherwise
+    for (const auto& op : order) {
+        if (ov::as_type_ptr<ov::op::v0::Parameter>(op)) {
```

Comment:
should we also track ReadValue ops?

Reply:
I found in the 2nd If branch
std::dynamic_pointer_cast<ov::op::util::VariableExtension>(op)) {

Reply:
No, we are only interested in the parameter dependant operations.
In steps 1 and 2 we are extracting the "main" graph -- the longest path from Parameter to Result, other graph paths will be ordered according to their connection to the "main" graph.

Please take a look at the [presentation](https://intel-my.sharepoint.com/:p:/p/evgeniia_nugmanova/EddCf8lCm9lFj87R2Xm0duQBZTJ4rb75tuB19G6cxWDTrw?e=jbNQFr) for a better understanding

Reply:
ok, I will, thanks!